homeassistant/components/icloud/config_flow.py

```diff
@@ -45,14 +43,10 @@ def __init__(self):
         self._trusted_device = None
         self._verification_code = None
 
-    def _configuration_exists(self, username: str, account_name: str) -> bool:
-        """Return True if username or account_name exists in configuration."""
+    def _configuration_exists(self, username: str) -> bool:
+        """Return True if username exists in configuration."""
         for entry in self._async_current_entries():
-            if (
-                entry.data[CONF_USERNAME] == username
-                or entry.data.get(CONF_ACCOUNT_NAME) == account_name
-                or slugify(entry.data[CONF_USERNAME].partition("@")[0]) == account_name
-            ):
+            if entry.data[CONF_USERNAME] == username:
```

Comment:
We can now set a unique id per config entry.

https://developers.home-assistant.io/docs/en/config_entries_config_flow_handler.html#unique-ids